### DISCUSSION TITLE
Does nvim-cmp depend on the builtin completion or can it be disabled?

### DISCUSSION THREAD

**Discussion by mikehaertl** [hrsh7th/nvim-cmp#941] [Q&A] [answered]
The setup instructions of the README have this line for [completeopt](https://neovim.io/doc/user/options.html#'completeopt'):

```vim
set completeopt=menu,menuone,noselect
```

From what I understand these settings relate to the builtin completion in vim. I wonder if this is really neccessary.

In my config I use `<c-j>` and `<c-k>` to navigate nvim-cmp completions. But I find myself often accidentally pressing `<c-n>` instead which seems to bring up the builtin vim completion in the background. So both (nvim-cmp + builtin completion) somehow conflict. I know that I could use the old default mappings - but that's not my point. I want to get used to `<c-j>` so `<c-n>` should have no mapping.

So I want to avoid this by completely disabling builtin completion via [complete](completely):

```vim
set complete=
set completeopt=
```

Is this a bad idea? Am I missing something where the builtin completion is still useful?

**Comment by Shougo** [comment, marked as answer]
You can disable built-in completion if not needed.
nvim-cmp does not depends on it.

**Reply by mikehaertl** [reply]
Cool, thanks for clarification.

So why is this configuration needed then? Is this for vim when the builtin popup menu system is used to render the menu? I.e. it's not needed on neovim?

**Reply by Shougo** [reply]
Because, plugins should not disable built in feature.
Other plugins may depend on it.

**Reply by mikehaertl** [reply]
I think you misunderstood. My question is why is this line required for nvim-cmp if it does not use the builtin completion:

https://github.com/hrsh7th/nvim-cmp/blob/433af3dffce64cbd3f99bdac9734768a6cc41951/README.md?plain=1#L62

What is this for?

**Reply by Shougo** [reply]
Oh, I get it.  It seems old.  It is not needed anymore.  Because it does not use native completion in default.

**Reply by mikehaertl** [reply]
I see, thanks. Maybe time to remove it then? 😉

**Reply by Shougo** [reply]
I think yes.

**Comment by megapixel-code** [comment]
Been searching for more than an hour on how to disable native completion, thank you !
